### PR TITLE
feat: Add TypeScript typings

### PR DESCRIPTION
**Pull request by mitsuruog**
Refer to #31.

I made a Typings file. but I just followed #31 way, though. 

I would happy if I can make the first contribution.
Thanks.

### CODE REVIEW

**Comment by jhnns** [comment]
Awesome, thanks. I'll review it the next days.

**Comment by DannyBrown** [comment]
This would be useful for me. Looks like all status checks pass. Did you get time to look at this @jhnns? Happy to help in any way possible 😄

File: src/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+declare namespace parseDomain {
+  interface IParsedDomain {
+    domain?: string;
+    subdomain?: string;
+    tld?: string;
+  }
+}
+
+declare function parseDomain (url: string, customTlds?: RegExp | string[], privateTlds?: boolean): parseDomain.IParsedDomain;
```
Comment (jhnns) [inline src/index.d.ts]: `customTlds` and `privateTlds` should be part of an `ParseOptions` object. Take a look at the README :)

**Comment by mitsuruog** [comment]
@jhnns I have fixed your review point. please review me again.

File: src/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  }
+
+  interface IParsedDomain {
+    domain?: string;
+    subdomain?: string;
+    tld?: string;
+  }
+}
+
+declare function parseDomain (url: string, options?: parseDomain.IParseOptions): parseDomain.IParsedDomain;
```
Comment (jhnns) [inline src/index.d.ts]: `parseDomain()` returns `null` when the given URL was not valid.

**Pull request merged**